src/pages/Fox/components/FoxFarming.tsx

```diff
@@ -0,0 +1,383 @@
+  } = useWallet()
+  const queryClient = useQueryClient()
+
+  const handleWalletModalOpen = useCallback(
+    () => dispatch({ type: WalletActions.SET_WALLET_MODAL, payload: true }),
+    [dispatch],
+  )
+
+  useEffect(() => {
+    appDispatch(marketApi.endpoints.findByAssetId.initiate(foxEthLpAssetId))
+  }, [appDispatch])
+
+  const foxEthMarketData = useAppSelector(state =>
+    selectMarketDataByAssetIdUserCurrency(state, foxEthLpAssetId),
+  )
+
+  const opportunityId = useMemo(
+    () =>
+      toOpportunityId({
+        assetNamespace: 'erc20',
+        assetReference: ETH_FOX_STAKING_EVERGREEN_CONTRACT,
+        chainId: assetAccountId ? fromAccountId(assetAccountId).chainId : ethChainId,
+      }),
+    [assetAccountId],
+  )
+
+  const stakingOpportunityByIdFilter = useMemo(() => {
+    const userStakingId = assetAccountId
+      ? serializeUserStakingId(assetAccountId, opportunityId)
+      : undefined
+    return { userStakingId, opportunityId }
+  }, [assetAccountId, opportunityId])
+
+  const opportunity = useAppSelector(state =>
+    selectStakingOpportunityByFilter(state, { stakingId: opportunityId }),
+  )
+
+  const userStakingOpportunity = useAppSelector(state =>
+    selectEarnUserStakingOpportunityByUserStakingId(state, stakingOpportunityByIdFilter),
+  )
+
+  const {
+    periodFinishQuery: {
+      data: periodFinish,
+      isLoading: isGetPeriodFinishQueryLoading,
+      isFetching: isGetPeriodFetchingLoading,
+    },
+  } = useFoxFarming(ETH_FOX_STAKING_EVERGREEN_CONTRACT)
+
+  const isGetPeriodFinishLoading = useMemo(
+    () => isGetPeriodFinishQueryLoading || isGetPeriodFetchingLoading,
+    [isGetPeriodFinishQueryLoading, isGetPeriodFetchingLoading],
+  )
+
+  const fetchOpportunityData = useCallback(async () => {
+    if (foxEthMarketData.price === '0') return
+
+    await appDispatch(
+      opportunitiesApi.endpoints.getOpportunityIds.initiate(
+        {
+          defiProvider: DefiProvider.EthFoxStaking,
+          defiType: DefiType.Staking,
+        },
+        { forceRefetch: true },
+      ),
+    )
+
+    appDispatch(
+      opportunitiesApi.endpoints.getOpportunityMetadata.initiate(
+        [
+          {
+            opportunityId,
+            defiType: DefiType.Staking,
+            defiProvider: DefiProvider.EthFoxStaking,
+          },
+        ],
+        { forceRefetch: true },
+      ),
+    )
+
+    if (assetAccountId) {
+      appDispatch(
+        opportunitiesApi.endpoints.getOpportunityUserData.initiate(
+          [
+            {
+              opportunityId,
+              accountId: assetAccountId,
+              defiType: DefiType.Staking,
+              defiProvider: DefiProvider.EthFoxStaking,
+            },
+          ],
+          { forceRefetch: true },
+        ),
+      )
+    }
+
+    return true
+  }, [assetAccountId, appDispatch, foxEthMarketData.price, opportunityId])
+
+  useEffect(() => {
+    queryClient.invalidateQueries({
+      queryKey: ['fetchOpportunityData'],
```

Comment:
paranoia: give this guy a more unique key just in case we ever reuse this and end up with collisions